roles/webserver/templates/unicorn.service.j2

```diff
@@ -18,7 +18,7 @@ PIDFile={{ unicorn_pid }}
 
 ExecStart={{ bundle_path }} exec unicorn -D -c {{ unicorn_config }} -E {{ rails_env }}
 ExecStop=/bin/kill -QUIT $MAINPID
-ExecReload=/bin/kill -HUP $MAINPID
+ExecReload=/bin/kill -USR2 $MAINPID
```

Comment:
I would add an extra task to forcefully restart in that case and delete it when the migration is complete.

Reply:
You don't think this reload would pick up a Ruby version change? I assumed it would. Can we delay that testing and adjusting until the next Ruby upgrade, as we'll have to check the upgrade process very carefully at that point anyway?

Reply:
@Matt-Yorkley You may be right, I haven't tested this. The manual says:

> You may replace a running instance of unicorn with a new one without losing any incoming connections. Doing so will reload all of your application code, Unicorn config, Ruby executable, and all libraries. The only things that will not change (due to OS limitations) are:

>  1. The path to the unicorn executable script. If you want to change to a different installation of Ruby, you can modify the shebang line to point to your alternative interpreter.

Since we are using rbenv, the path could be `~/.rbenv/shims/unicorn` which won't change. In that case we are good. But if the OS supplies the path of the currently running script, it could give us `~/.rbenv/versions/2.3.7/bin/unicorn`. I don't know how that's determined.

Let's assume it's good and test well for the next Ruby upgrade. We just have to be aware that if we upgrade Ruby without upgrading unicorn at the same time, the executable would be the same and we may not notice the difference until we upgrade unicorn and still have the old version running.